packages/server/lib/server-base.ts

```diff
@@ -436,20 +503,39 @@ export abstract class ServerBase<TSocket extends SocketE2E | SocketCt> {
     //     domain: "google"
     //   }
     // }
+    const props = this._getCurrentSecondaryOrigin() || this._getDefaultRemoteState()
 
-    const props = _.extend({}, {
+    debug('Getting remote state: %o', props)
+
+    return props
+  }
+
+  _getDefaultRemoteState (): Cypress.RemoteState {
+    return _.extend({}, {
       auth: this._remoteAuth,
       props: this._remoteProps,
       origin: this._remoteOrigin,
       strategy: this._remoteStrategy,
-      visiting: this._remoteVisitingUrl,
       domainName: this._remoteDomainName,
       fileServer: this._remoteFileServer,
     }) as Cypress.RemoteState
+  }
 
-    debug('Getting remote state: %o', props)
+  _addRemoteState (url: string, options: { auth?: {}} = {}) {
```

Comment:
Is it possible for the same remote state to already exist and we just blow it away in the map? Wondering is something like `_setRemoteStateFor` or whatnot is more appropriate, but I might be overthinking it.

Reply:
Renamed to `set` in the new `RemoteStates` class.